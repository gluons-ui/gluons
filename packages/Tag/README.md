## Mandala UI Tag Component
Mandala is a React component library that utilizes a declarative CSS library for styling.

### Installation
Check the [MonoRepo README](https://github.com/mandala-ui/mandala) for installation.

### Description
Tag is just a simple element to show metadata in a small font

### Usage
if using the whole library:
```js
import { Tag } from 'mandala';
```
if only using this component:
```js
import Tag from '@mandala-ui/split-button';
```
### Props
* `backgroundColor` - the color of the surface area of the switches
* `disabled` - sets the button as disabled and reduces opacity 40%
* `isOn` - prop to switch which side of the switch is on
* `offColor` - the color of the off section when the switch is off
* `onClick` - function for onClick
* `onColor` - the color of the on section when the switch is on
* `pill` - sets the radius of the ends to 100%
* `radius` - the border radius of the button (1-4, clamped), is disabled if `pill` is set to `true`
* `showLabels` - shows the "on" and "off" text labels

| propName | propType | defaultValue | isRequired |
| -------- | -------- | ------------ | ---------- |
| backgroundColor | string | 'white' | -          |
| disabled  | string | 'black'      | -          |
| isOn      | boolean| false        | -          |
| offColor  | string | 'black'      | -          |
| onClick   | function| null        | -          |
| onColor   | string | 'black'      | -          |
| pill      | boolean| false        | -          |
| radius    | boolean| false        | -          |
| showLabels| boolean| false        | -          |


#### TODO:
- [ ] callback for analytics, etc
- [ ] colored border
- [ ] border width

![mandala-bottom](https://user-images.githubusercontent.com/1824267/38281372-32114e5c-375f-11e8-9fbe-e5191b96429c.png)
