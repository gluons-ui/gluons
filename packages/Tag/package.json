{
  "author": "Lucas K Allmon",
  "dependencies": {
    "lodash.clamp": "^4.0.3"
  },
  "description": "Mandala UI Tag Component",
  "keywords": [
    "react",
    "css",
    "library",
    "components",
    "interface",
    "functional",
    "OOCSS",
    "declarative",
    "ui",
    "user",
    "ux"
  ],
  "license": "MIT",
  "main": "lib/Tag.js",
  "module": "es/Tag.js",
  "name": "@mandala-ui/tag",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "prepare": "nwb build-react-component"
  },
  "version": "1.0.1"
}
