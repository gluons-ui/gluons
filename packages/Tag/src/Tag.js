import React from 'react';
import PropTypes from 'prop-types';

const Tag = ({
  children,
  color,
  large }) => {
  const isLarge = `${large ? 'f5 ph3 pv2' : 'f7 ph2 pv1'}`;
  return (
    <div className="tag">
      <span className={`bg-${color} br-pill dib ${isLarge} white`}>
        {children || null}
      </span>
    </div>
  );
};

Tag.propTypes = {
  color: PropTypes.string,
  children: PropTypes.node,
  large: PropTypes.bool,
};

Tag.defaultProps = {
  children: null,
  color: 'blue',
  large: false,
};

export default Tag;
