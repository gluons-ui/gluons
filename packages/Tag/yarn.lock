# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


lodash.clamp@^4.0.3:
  version "4.0.3"
  resolved "https://registry.yarnpkg.com/lodash.clamp/-/lodash.clamp-4.0.3.tgz#5c24bedeeeef0753560dc2b4cb4671f90a6ddfaa"
