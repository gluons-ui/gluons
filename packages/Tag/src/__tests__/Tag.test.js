import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import Tag from '../Tag';

configure({ adapter: new Adapter() });

describe('<Tag />', () => {
  const component = shallow(<Tag />);

  it('renders defaults correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('increases the size if large prop is passed', () => {
    expect(shallow(<Tag large />).find('.tag').hasClass('f5 ph3 pv2')).toEqual(true);
  });
});
