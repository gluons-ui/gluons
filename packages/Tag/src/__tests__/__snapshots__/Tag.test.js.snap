// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> increases the size if large prop is passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tag
    color="blue"
    large={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="bg-blue br-pill dib f5 ph3 pv2 white"
      />,
      "className": "tag",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": null,
        "className": "bg-blue br-pill dib f5 ph3 pv2 white",
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="bg-blue br-pill dib f5 ph3 pv2 white"
        />,
        "className": "tag",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "className": "bg-blue br-pill dib f5 ph3 pv2 white",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<Tag /> renders defaults correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tag
    color="blue"
    large={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="bg-blue br-pill dib f7 ph2 pv1 white"
      />,
      "className": "tag",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": null,
        "className": "bg-blue br-pill dib f7 ph2 pv1 white",
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="bg-blue br-pill dib f7 ph2 pv1 white"
        />,
        "className": "tag",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "className": "bg-blue br-pill dib f7 ph2 pv1 white",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
